### PR TITLE
#47 - Ping mission makers on discord

### PR DESCRIPTION
**Pull request by TomBurch**

### CODE REVIEW

File: app/Discord/ChannelEnum.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace App\Discord;
+
+class ChannelEnum
```
Comment (TomBurch) [inline app/Discord/ChannelEnum.php:5]: Can be replaced by actual Enums when we upgrade to php 8

**Comment by TomBurch** [comment]
All non-discord related changes are from https://github.com/ARCOMM/ARCHUB/pull/69 which is required for datastore

**Comment by TomBurch** [comment]
At the moment it's set to use a non-archub google cloud - if I'm given access to that one then it can be changed
Currently mission makers would have to be manually added to the datastore to opt-in, but I plan on adding a command to the discord bot for opt-in/out
image

File: app/Discord/DiscordWebhook.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        {
+            throw new Exception("Webhook not found");
+        }
+    }
+
+    private static function getDiscordIdFromUser(User $user)
+    {
+        $query = self::Datastore()->query()
+        ->kind('DiscordIdentifier')
+        ->filter('SteamID', '=', (int)$user->steam_id)
```
Comment (BorderKeeper) [inline app/Discord/DiscordWebhook.php:82]: How does this work? Does discord have to be linked with steam?

File: app/Models/JoinRequests/JoinRequest.php [hunk 3]
```diff
@@ -60,12 +60,12 @@ public function source()
      */
     public static function items($s = '', $o = 'desc')
     {
-        $order = Input::get('order', $o);
+        $order = Request::input('order', $o);
```
Comment (BorderKeeper) [inline app/Models/JoinRequests/JoinRequest.php:63]: frowny face

**Review by BorderKeeper — APPROVED** [review]
lgtm with the caveat of having to add links manually

**Comment by TomBurch** [comment]
Replaced by https://github.com/ARCOMM/ARCHUB/pull/77

**Pull request closed without merging**